sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -438,4 +421,32 @@ object UnsupportedOperationChecker extends Logging {
     throw new AnalysisException(
       msg, operator.origin.line, operator.origin.startPosition, Some(operator))
   }
+
+  private def checkForStreamStreamJoinWatermark(join: Join): Unit = {
+    val watermarkInJoinKeys = StreamingJoinHelper.isWatermarkInJoinKeys(join)
+
+    val hasValidWatermarkRange = join.joinType match {
```

Comment:
Let's also keep the comment https://github.com/apache/spark/pull/30395/files#diff-7c879c08d2f379c139d5229a88857229ae69bb48f0a138a3d64e1b2dde3502feL341 in this new method.

Reply:
@xuanyuanking - added back. sorry I was missing that in the first place.